Jenkinsfile

```diff
@@ -1,175 +1,60 @@
-        variable: 'SAUCELABS_API_KEY']]) {
-        withEnv(["PYTEST_ADDOPTS=${PYTEST_ADDOPTS} " +
-          "-n=${processes} " +
-          "--base-url=${PYTEST_BASE_URL} " +
-          "--driver=SauceLabs " +
-          "--variables=capabilities.json " +
-          "--color=yes"]) {
-          sh "tox -e ${environment}"
-        }
+      try {
+        config = readYaml file: "jenkins.yml"
       }
-    }
-  } catch(err) {
-    currentBuild.result = 'FAILURE'
-    throw err
-  } finally {
-    dir('results') {
-      stash environment
-    }
-  }
-}
-
-/** Send a notice to #fxtest-alerts on irc.mozilla.org with the build result
- *
- * @param result outcome of build
-*/
-def ircNotification(result) {
-  def nick = "fxtest${BUILD_NUMBER}"
-  def channel = '#fx-test-alerts'
-  result = result.toUpperCase()
-  def message = "Project ${JOB_NAME} build #${BUILD_NUMBER}: ${result}: ${BUILD_URL}"
-  node {
-    sh """
-        (
-        echo NICK ${nick}
-        echo USER ${nick} 8 * : ${nick}
-        sleep 5
-        echo "JOIN ${channel}"
-        echo "NOTICE ${channel} :${message}"
-        echo QUIT
-        ) | openssl s_client -connect irc.mozilla.org:6697
-    """
-  }
-}
+      catch (e) {
+        config = []
+      }
+      println "config ==> ${config}"
 
-stage('Checkout') {
-  node {
-    timestamps {
-      deleteDir()
-      checkout scm
-      stash 'workspace'
+      if (!config || (config && config.pipeline && config.pipeline.enabled == false)) {
+        println "Pipeline disabled."
+      }
     }
-  }
-}
 
-def builders = [:]
-for (entry in entrySet(environments)) {
-  def environment = entry.key
-  def capabilities = entry.value
-  builders[(environment)] = {
-    node {
-      timeout(time: 1, unit: 'HOURS') {
-        timestamps {
-          deleteDir()
-          unstash 'workspace'
-          writeCapabilities(capabilities)
-          withCredentials([[
-            $class: 'FileBinding',
-            credentialsId: 'SUMO_VARIABLES',
-            variable: 'VARIABLES']]) {
-            withEnv(["PYTEST_ADDOPTS=--variables=${VARIABLES}"]) {
-              runTox(environment)
-            }
-          }
-        }
+    docker_image = "${config.project.docker_name}:${GIT_COMMIT_SHORT}"
+
+    stage("Build") {
+      if (!dockerImageExists(docker_image)) {
+            sh "echo 'ENV GIT_SHA ${GIT_COMMIT}' >> Dockerfile"
+
+            // Docker always copies data as root. Chowning files to root before
+            // the build command will prevent docker cache invalidation due to
+            // different file metadata. When we upgrade to a newer docker
+            // version we should probably use --chown flag for COPY. See
+            // https://github.com/moby/moby/pull/34263
+            dockerRun('debian', [
+                    "docker_args": "-v `pwd`:/app",
+                    "cmd": "cd /app && chown -c root.root -R requirements bower.json package.json && chmod a+rw -R requirements bower.json package.json",
```

Comment:
@pmac fiy i updated this command to 
1. first chown to root to fix the docker bug
2. chmod to a+rw to make jenkins able to alter the files on next build